src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs

```diff
@@ -49,7 +50,9 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
         {
             var keyBuilder = new StringBuilder();
             var routeValues = context.ActionContext.RouteData.Values;
-            var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
+            var controller = RouteUtility.GetNormalizedRouteValue(
+                RazorViewEngine.ControllerKey,
+                context.ActionContext);
 
             // format is "{viewName}:{isPartial}:{controllerName}:{areaName}:"
             keyBuilder.Append(context.ViewName)
```

Comment:
It's developer supplied as opposed to application user \ browser supplied. Changing the cache key doesn't make a significant difference if the value ends up being the wrong case resulting a failure in lookup.